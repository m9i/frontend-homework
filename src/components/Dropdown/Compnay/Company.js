import React from "react"
import { selectCompanyAction } from "../../../store/actions/selectCompanyAction"
import { connect } from "react-redux"
import { toggleDropdownAction } from "../../../store/actions/toggleDropdownAction"

const Company = ({ onCompanyChange, selectedCompany, company, hideDropdown }) => {
  const onCompanyChanged = () => {
    onCompanyChange(company)
    hideDropdown()
  }

  return (
    <>
      <div
        className={
          company.value === selectedCompany.value ? "selected-company" : null
        }
        onClick={() => onCompanyChanged()}
      >
        <span> {company.value}</span>
        {company.value === selectedCompany.value && (
          <i className="material-icons-outlined nav__list__item--icon">done</i>
        )}
      </div>
    </>
  )
}

const mapStateToProps = (state) => ({
  selectedCompany: state.company.selectedCompany,
})

const mapDispatchToProps = (dispatch) => ({
  onCompanyChange: (item) => dispatch(selectCompanyAction(item)),
  hideDropdown: () => dispatch(toggleDropdownAction(false)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Company)
