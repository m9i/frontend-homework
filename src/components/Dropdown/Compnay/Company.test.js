import React from "react";
import Company from "./Company";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

test("Company snapshot", () => {
  const component = renderer.create(<Company />);
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
