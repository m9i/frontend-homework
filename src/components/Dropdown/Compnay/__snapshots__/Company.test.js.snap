// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company snapshot 1`] = `<Connect(Company) />`;

exports[`Render Company snapshot 1`] = `<Connect(Company) />`;
